include/numerics/sparse_matrix.h

```diff
@@ -406,6 +406,17 @@ class SparseMatrix : public ReferenceCountedObject<SparseMatrix<T>>,
    */
   virtual void get_transpose (SparseMatrix<T> & dest) const = 0;
 
+  /**
+   * Get a row from the matrix
+   * @param i The matrix row to get
+   * @param indices A container that will be filled with the column indices
+   *                corresponding to (possibly) non-zero values
+   * @param values A container holding the column values
+   */
+  virtual void get_row(numeric_index_type /*i*/,
+                       std::vector<numeric_index_type> & /*indices*/,
+                       std::vector<T> & /*values*/) const {}
```

Comment:
Stick a libmesh_not_implemented() in the base class to make sure anyone accidentally using it gets an error rather than silently-mistaken values?

Reply:
Could you add this before we merge?

Reply:
I did...

Reply:
Sorry, I didn't notice that GitHub now just adds a tiny "outdated" tag rather than collapsing/hiding conversations about outdated code entirely.